hibernate-core/src/main/java/org/hibernate/query/sqm/tree/domain/SqmEmbeddedValuedSimplePath.java

```diff
@@ -77,6 +79,30 @@ public SqmPath<?> resolvePathPart(
 		return sqmPath;
 	}
 
+	@Override
+	public SqmPath<?> get(String attributeName) {
+		final DomainType<?> domainType = getRoot( getLhs() ).getReferencedPathSource().getSqmPathType();
```

Comment:
It seems to me that a `getRoot` here is not correct. Can you explain why you did that? Also, it would be nice if we could somehow capture the fact that a path source is generic to only do the `findGenericEmbeddableAttribute` if necessary.

Reply:
At this point `getRoot` was needed because we would be resolving not the "first level" embeddable but one of its sub-properties.

I've reworked the logic to be handled by `EmbeddedSqmPathSource` to be able to retrieve the correct type for the "first level" embeddable attribute. Also I added the `generic` flag like you suggested.

Reply:
Moving the logic from `SqmEmbeddedValuedSimplePath` to `EmbeddedSqmPathSource` might cause some trouble with the assertions I asked you to add, because the `EmbeddedSqmPathSource` is what `SqmEmbeddedValuedSimplePath` exposes as "type" information AFAICT, but only some additional assertions will be able to ensure we don't expose any generic info through JPA contracts.